tests/infra_test.go

```diff
@@ -665,117 +669,116 @@ var _ = Describe("Infrastructure", func() {
 				Transport: tr,
 			}
 
-			errors := make(chan error)
-			for ix := 0; ix < concurrency; ix++ {
-				go func(ix int) {
-					req, _ := http.NewRequest("GET", metricsURL, nil)
-					resp, err := client.Do(req)
+			for _, url := range metricsURLs {
+				if urlIsIPv6(url) {
+					By("Skip Scraping the Prometheus endpoint (IPv6) until issue https://github.com/kubevirt/kubevirt/issues/4145 is fixed")
+					continue
+				}
+
+				By(fmt.Sprintf("Scraping the Prometheus endpoint %s", url))
+				errors := make(chan error)
+				for ix := 0; ix < concurrency; ix++ {
+					go func(ix int) {
+						req, _ := http.NewRequest("GET", url, nil)
+						resp, err := client.Do(req)
+						if err != nil {
+							fmt.Fprintf(GinkgoWriter, "client: request: %v #%d: %v\n", req, ix, err) // troubleshooting helper
+						} else {
+							resp.Body.Close()
+						}
+						errors <- err
+					}(ix)
+				}
+
+				errorCount := 0
+				for ix := 0; ix < concurrency; ix++ {
+					err := <-errors
 					if err != nil {
-						fmt.Fprintf(GinkgoWriter, "client: request: %v #%d: %v\n", req, ix, err) // troubleshooting helper
-					} else {
-						resp.Body.Close()
+						if urlErr, ok := err.(*netURL.Error); ok {
+							Expect(urlErr.Err.Error()).ToNot(ContainSubstring("connect: network is unreachable"))
+							Expect(urlErr.Err.Error()).To(ContainSubstring("Client.Timeout exceeded while awaiting headers"))
+						}
+						errorCount += 1
```

Comment:
Shoudun't the `errorCount` increase be inside the `if`?

Reply:
no, because its already under `if err != nil {`
we want to know on errors in case `if urlErr, ok := err.(*netURL.Error); ok {` failed
or even better to assert if it happens
and even better to do `(we can skip sending to the channel if err == nil btw)`
which was written here
https://github.com/kubevirt/kubevirt/pull/4140#discussion_r488635209

Reply:
Added assert on the parsing